stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -191,6 +191,12 @@ end
 (*)(D::Diagonal, x::Number) = Diagonal(D.diag * x)
 (/)(D::Diagonal, x::Number) = Diagonal(D.diag / x)
 (\)(x::Number, D::Diagonal) = Diagonal(x \ D.diag)
+(^)(D::Diagonal, a::Number) = Diagonal(D.diag .^ a)
+(^)(D::Diagonal, a::Real) = Diagonal(D.diag .^ a) # for disambiguation
+(^)(D::Diagonal, a::Integer) = Base.literal_pow(^, D, Val(a)) # for disambiguation
```

Comment:
Is this a good idea?
```
julia> p = 3; @code_warntype Diagonal([1.0, 2.0])^p
MethodInstance for ^(::Diagonal{Float64, Vector{Float64}}, ::Int64)
  from ^(D::Diagonal, a::Integer) in LinearAlgebra at REPL[14]:2
Arguments
  #self#::Core.Const(^)
  D::Diagonal{Float64, Vector{Float64}}
  a::Int64
Body::Any
1 ─ %1 = Base.literal_pow::Core.Const(Base.literal_pow)
│   %2 = LinearAlgebra.Val(a)::Val
│   %3 = (%1)(LinearAlgebra.:^, D, %2)::Any
└──      return %3
```

```suggestion
(^)(D::Diagonal, a::Integer) = Diagonal(D.diag .^ a) # for disambiguation
```